core/src/main/java/com/novoda/noplayer/exoplayer/mediasource/ExoPlayerTrackSelector.java

```diff
@@ -4,36 +4,66 @@
+import com.novoda.utils.Optional;
 
 public class ExoPlayerTrackSelector {
 
     private final DefaultTrackSelector trackSelector;
+    private final RendererTrackIndexExtractor rendererTrackIndexExtractor;
+
+    public static ExoPlayerTrackSelector newInstance(DefaultTrackSelector trackSelector) {
+        RendererTrackIndexExtractor rendererTrackIndexExtractor = new RendererTrackIndexExtractor();
+        return new ExoPlayerTrackSelector(trackSelector, rendererTrackIndexExtractor);
+    }
 
-    public ExoPlayerTrackSelector(DefaultTrackSelector trackSelector) {
+    ExoPlayerTrackSelector(DefaultTrackSelector trackSelector, RendererTrackIndexExtractor rendererTrackIndexExtractor) {
         this.trackSelector = trackSelector;
+        this.rendererTrackIndexExtractor = rendererTrackIndexExtractor;
     }
 
-    TrackGroupArray getAudioTrackGroups() {
-        return trackInfo().getTrackGroups(TRACK_TYPE_AUDIO);
+    TrackGroupArray trackGroups(TrackType trackType, RendererTypeRequester rendererTypeRequester) {
+        Optional<Integer> audioRendererIndex = rendererTrackIndexExtractor.extract(trackType, mappedTrackInfoLength(), rendererTypeRequester);
+        return audioRendererIndex.isAbsent() ? TrackGroupArray.EMPTY : trackInfo().getTrackGroups(audioRendererIndex.get());
     }
 
-    private MappingTrackSelector.MappedTrackInfo trackInfo() {
+    public void clearSelectionOverrideFor(TrackType trackType, RendererTypeRequester rendererTypeRequester) {
+        Optional<Integer> audioRendererIndex = rendererTrackIndexExtractor.extract(trackType, mappedTrackInfoLength(), rendererTypeRequester);
+        if (audioRendererIndex.isPresent()) {
+            trackSelector.clearSelectionOverrides(audioRendererIndex.get());
+        }
+    }
+
+    public ExoPlayerMappedTrackInfo trackInfo() {
         MappingTrackSelector.MappedTrackInfo trackInfo = trackSelector.getCurrentMappedTrackInfo();
 
         if (trackInfo == null) {
             throw new NullPointerException("Track info is not available.");
         }
-        return trackInfo;
+        return new ExoPlayerMappedTrackInfo(trackInfo);
+    }
+
+    private int mappedTrackInfoLength() {
+        return trackSelector.getCurrentMappedTrackInfo().length;
     }
 
-    void setSelectionOverride(TrackGroupArray trackGroups, MappingTrackSelector.SelectionOverride selectionOverride) {
-        trackSelector.setSelectionOverride(TRACK_TYPE_AUDIO, trackGroups, selectionOverride);
+    void setSelectionOverride(TrackType trackType,
+                              RendererTypeRequester rendererTypeRequester,
+                              TrackGroupArray trackGroups,
+                              MappingTrackSelector.SelectionOverride selectionOverride) {
+        Optional<Integer> audioRendererIndex = rendererTrackIndexExtractor.extract(trackType, mappedTrackInfoLength(), rendererTypeRequester);
+        if (audioRendererIndex.isPresent()) {
```

Comment:
This being absent is a case that should not happen right? I think w should warn the caller it didn't work so that we can bubble up until we can send an error though the ErrorListener.